examples/try_deeprec.py

```diff
@@ -0,0 +1,28 @@
+
```

Comment:
Could you please transform this python file in a notebook as a quickstart, to follow the structure that we already have. Similar to this: https://github.com/Microsoft/Recommenders/blob/staging/notebooks/00_quick_start/sar_single_node_movielens.ipynb

Reply:
Also could you use a public data set such as Movielens or Criteo instead of a synthetic one? 
You could choose to do a quick start notebook or a deep dive one (see Recommenders/notebooks/02_model/). In both cases the data size should not be very large, so that the notebook runs in a few minutes (for Criteo you could subsample the data). The difference is that in the deep dive notebooks you explain in more detail what happens at each step and how the algorithm works under the hood.

Reply:
@Leavingseason For this notebook, do you think you can use Movielensm or do you prefer Criteo instead? If the latter, one of the team should write a dataset loader for Criteo first.

Reply:
@anargyri deeprec uses a different type of dataloaders, I put them under the IO folder. I can try using a small sample of Criteo dataset.

Reply:
Criteo would be great, not sure if @yexing99 has worked with it. @wutaomsft FYI

Reply:
The approach we follow in the repo is to avoid external dataset loaders and instead use data set loaders common to all algorithms, as in https://github.com/Microsoft/Recommenders/blob/master/reco_utils/dataset/movielens.py.
These return a pandas dataframe with the specified columns. This dataframe can then be manipulated with the external libraries and algorithms.

So, I suggest a similar approach here, in which you take the pandas dataframe that we provide and you apply the data transformations from your code to it. It may look like a detour, but this enables everyone to easily apply DeepRec on any new dataset they have: they just need to write a new loader under reco_utils/dataset/ and they don't need to modify your code. Moreover, this loader will be reusable by the other algorithms we have in the repo.

Reply:
I don't think so. DeepRec contains different types of data, e.g. FFM format for xdeepfm,  DKN format which requires aligned news words and news entities. It's hard to use dataframe to store them. Besides the data format, in feature-based algos the feature files are usually large (in production, may be dozens of GB). We can never try to load the entire feature file into memory via dataframe .

Reply:
I see. Is FFM format the same as in the FFM paper and in https://github.com/guestwalk/libffm ?
Would it make sense to have a loader for FFM, one for DKN etc.? I am thinking that having the data in sparse FFM format may be useful in its own right, e.g. the user can do some data preprocessing, visualization, input the data to an FFM etc.

Reply:
Yes, the format is almost the same as in your link. In fact we are indeed using an iterator for FFM, and an iterator for DKN.  We hope that similar feature-aware models can share the same type of loader (iterator), but not a unified loader which can take all types of feature-aware data (for example, not good for one loader which can take both FFM and DKN format)